File: pset6/readability/readability.py
# program to get the grade level of text
from cs50 import get_string
# getting user input
text = get_string("Text: ")
# i nitializing variables:
letters = words = sentences = i = 0

# countgin the no of letters words and sentencs
while i < len(text):
    if text[i].isalpha():
        letters += 1
# counting words
    if i == 0 and text[i] != " " or i != 0 and text[i] == " ":
        words += 1
# counting sentences/
    if text[i] == "." or text[i] == "!" or text[i] == "?":
        sentences += 1
    i += 1

L = (letters / words) * 100
S = (sentences / words) * 100
index = round(0.0588 * L - 0.296 * S - 15.8)

# printing
if index < 1:
    print("Before Grade 1")
elif index >= 16:
    print("Grade 16+")
else:
    print(f"Grade {index}")

File: pset2/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <ctype.h>
#include <string.h>
#include <math.h>

//Delcaring the prototype;
int count_letters(string word);
int count_words(string word);
int count_sentences(string word);

int main(void)
{
    //getting user input;
    string input = get_string("Text: ");
    float letters = count_letters(input);
    float words = count_words(input);
    float L = ((letters / words) * 100);
    float sentences = count_sentences(input);
    float S = ((sentences / words) * 100);
    // coleman lieu index formula;
    int index = round(0.0588 * L - 0.296 * S - 15.8);
    //printing the grade;
    if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }

}
//fnction for the count of letters;
int count_letters(string word)
{
    int letters = 0;

    for (int i = 0, len = strlen(word); i < len; i++)
    {
        if (isalpha(word[i])) //checking for the alphabet;
        {
            letters++;
        }
    }
    return letters;
}

// count the no of words
int count_words(string word)
{
    int words = 0;
    if (strlen(word) > 1)
    {
        words = 1;
    }
    for (int i = 0, len = strlen(word); i < len; i++)
    {
        if (isspace(word[i])) //checking white spaces
        {
            words++;
        }
    }
    return words;
}
// coutnig the no of sentences
int count_sentences(string word)
{
    int sentences = 0;
    for (int i = 0, len = strlen(word); i < len; i++)
    {
        if (word[i] == '?' || word[i] == '!' || word[i] == '.')
        {
            sentences++;
        }
    }
    return sentences;
}